src/rf/apps/workers/process.py

```diff
@@ -148,18 +118,19 @@ def _emr_hand_off(self, data):
         # return False if it fails to start.
         # EMR posts directly to queue upon competion.
 
-        layer_id = data['layer_id']
-        layer_images = LayerImage.objects.filter(layer_id=layer_id)
-        reprojected_images = LayerImage.objects.filter(
-            layer_id=layer_id,
-            status=enums.STATUS_REPROJECTED)
-        ready_to_process = len(layer_images) == len(reprojected_images)
+        layer_id = None
+        try:
+            layer_id = data['layer_id']
+            layer_images = LayerImage.objects.filter(layer_id=layer_id)
+            valid_images = LayerImage.objects.filter(layer_id=layer_id,
+                                                     status=enums.STATUS_VALID)
+            ready_to_process = len(layer_images) == len(valid_images)
+        except (LayerImage.DoesNotExist, KeyError):
```

Comment:
I think we need to overall separate exceptions caused by parsing `data` and DB related exceptions.